### PR TITLE
Don't remove empty annotations

### LINKED ISSUE #5823: Deleting a notebook entry without tags throws an error and the modal does not close
#### Summary

When deleting a notebook entry without a tag, an unhandled error is thrown in the console and the delete modal does not close as expected.

#### Expected vs Current Behavior

Expected: The entry should be deleted and the modal should close.
Current: The entry is deleted, but the modal does not close and an error can be seen in the console

#### Steps to Reproduce

1. Create a notebook
2. Add an entry
3. Add another entry and tag it
4. Delete the entry without tags
5. Observe an error in the console and the modal does not close
6. Delete the entry with tags
7. The entry deletes and modal closes as expected with no errors

#### Environment

```
Version: 2.1.1
Build Date: Fri Sep 30 2022 15:41:32 GMT-0700 (Pacific Daylight Time)
Revision: 1c00f1a731263cd7573f02e0a8f1591b634d2298
Branch: release/2.1.1
```

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information

image

### PR DESCRIPTION
**Pull request by scottbell**
Closes #5823 

### Describe your changes:
Check if annotations are empty before sending it along to the API.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by scottbell** [comment]
Now checking that annotations exist before trying to delete them:

https://user-images.githubusercontent.com/9853862/193394327-ec777372-fc0b-48ac-8972-21d5be907bc0.mov

File: package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "moment-timezone": "0.5.37",
     "nyc": "15.1.0",
     "painterro": "1.2.78",
-    "playwright-core": "1.26.1",
+    "playwright-core": "1.25.2",
```
Comment (scottbell) [inline package.json:54]: playwright-test was complaining core wasn't aligned

**Comment by ozyx** [comment]
Hate to be the bearer of *slightly* inconvenient news but this needs to be rebased onto `release/2.1.1` and retargeted

**Comment by scottbell** [comment]
Closing to set target to `release/2.1.1`

**Comment by scottbell** [comment]
> Hate to be the bearer of _slightly_ inconvenient news but this needs to be rebased onto `release/2.1.1` and retargeted

D'oh! Thanks for the catch 😀 I've reissued this PR here:
https://github.com/nasa/openmct/pull/5829

**Pull request closed without merging**